sympy/matrices/tests/test_matrices.py

```diff
@@ -2467,3 +2467,28 @@ def test_doit():
     a = Matrix([[Add(x,x, evaluate=False)]])
     assert a[0] != 2*x
     assert a.doit() == Matrix([[2*x]])
+
+def test_issue_9457():
+    # for row_del(index)
+    M = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    M.row_del(1)
+    assert M == Matrix([[1, 2, 3], [3, 4, 5]])
+    N = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    N.row_del(-2)
+    assert N == Matrix([[1, 2, 3], [3, 4, 5]])
+    P = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    raises(ValueError, lambda: P.row_del(10))
```

Comment:
It's better if the error message quotes the index value and the valid minimum/maximum, this can give the user valuable hints about the cause of the error.
However, if other similar error messages don't do that, you don't need to go that extra mile (but I'd still like it if you did :-) )